### ISSUE TITLE
`$env.FILE_PWD` for sourced-scripts

### ISSUE THREAD

**Issue by hyperupcall** [nushell/nushell#10626]
### Related problem

Similar to `BASH_SOURCE`, I would like [a way](https://unix.stackexchange.com/questions/4650/how-to-determine-the-path-to-a-sourced-tcsh-or-bash-shell-script-from-within-the) to get to the directory of the currently-sourced file.

### Describe the solution you'd like

$ bash -c 'pwd'
/home/me
$ bash -c 'echo $env.FILE_PWD > f.nu'
$ source f.nu
/home/me/f.nu

### Describe alternatives you've considered

`FILE_PWD` doesn't currently work this way. I have searched the issue queue for `BASH_SOURCE` and persed the docs, but found nothing.

**Comment by sholderbach** [comment]
Related but slightly different #9776

**Comment by fdncred** [comment]
I looked into this a little bit.

Example script
```nushell
❯ cat test.nu
echo $"FILE_PWD: ($env.FILE_PWD)"
echo $"CURRENT_FILE: ($env.CURRENT_FILE)"
```

Run with nu
```
❯ nu test.nu
FILE_PWD: D:\fz
CURRENT_FILE: D:\fz\test.nu
```

Run with source-env
```
❯ nu test.nu
FILE_PWD: D:\fz
CURRENT_FILE: D:\fz\test.nu
```

Run with source
```
❯ source test.nu
Error: nu::shell::column_not_found

  × Cannot find column
   ╭─[D:\fz\test.nu:1:1]
 1 │ echo $"FILE_PWD: ($env.FILE_PWD)"
   ·                   ──┬─ ────┬───
   ·                     │      ╰── cannot find column 'FILE_PWD'
   ·                     ╰── value originates here
 2 │ echo $"CURRENT_FILE: ($env.CURRENT_FILE)"
   ╰────
```

Theoretically `source` could be modified to work like `source-env` wrt these env vars. I'm just not sure if that's something we want to do.

**Comment by hyperupcall** [comment]
> I'm just not sure if that's something we want to do.

Do you wish to avoid implementing this because there are certain side effects (that I do not know about)? Or that I'm going down a bad path and there is a generally better way of doing things that allows me to `$env.FILE_PWD`?

**Comment by fdncred** [comment]
Neither, I just don't know. It'll take someone else on the core-team to chime in.

**Comment by amtoine** [comment]
i understand this as follows:
- it makes sense to have `$env.FILE_PWD` and `$env.CURRENT_FILE` in a script, because a script has a *static* location on the filesystem, i.e. if you don't change the `$env.PATH`, `which foo.nu` should always point the the same location
- with a module, you `use foo.nu` and then the definitions are brought into scope but it makes less sense to *remember the location* of the original module because you can move around in Nushell, even move the module on disk and it won't affect the useability of the module loaded in memory
> **Note**
> this is why i opened https://github.com/nushell/nushell/issues/9776 in the first place, because `$env.FILE_PWD` and `$env.CURRENT_FILE` don't make too much sense in the context of a module
- i guess now, when we come to `source`ing scripts, it's a bit like with a module: you can `source foo.nu`, that'll bring all the definitions inside `foo.nu` to the current scope, then you can move the file around in the filesystem and it won't bring any issues when it comes to running the original definitions from the original `foo.nu` because they are loaded in memore already

i'm trying to put words on the intuition of @fdncred from what i understand, please correct me if i'm heading in the wrong direction :relieved: 

**TL;DR**:
- a script doesn't move on disk when you `nu foo.nu` => it makes sense to be able to point to the location of the script from inside it with `$env.CURRENT_FILE` and `$env.FILE_PWD`
- a module or sourcing a script: the definitions are brought into the current scope but then you can move the original files around => you cannot really point to the location of a module and a script that has been sourced

**Comment by fdncred** [comment]
@amtoine The one difference that I'd add to your TL;DR is that `source`ing a file can also run the contents.

For example:
Filename: test.nu
contents:
```nushell
def fun [] {
  echo Hi!
}
fun
```

So, in this instance, it seems like having the env vars available would be helpful.

**Comment by amtoine** [comment]
ooh yeah true @fdncred, maybe the is nuance to add to the `source`ing of scripts then :+1:

**Comment by CAD97** [comment]
*Technically* `use`ing a module can run code as well:

<details><summary><code>x.nu</code></summary>

```nushell
export-env {
    print "Thank you for using x.nu!"
}
```
</details>

```shell
~> use x.nu
Thank you for using x.nu!
~> std help x
Help pages from external command x:
Module: x

This module exports environment.
{
    print "Thank you for using x.nu!"
}
```

**Comment by cptpiepmatz** [comment]
We have the `path self` for that now, it runs at parse time and returns a string to the path file path.
